dsl_parser/import_resolver/default_import_resolver.py

```diff
@@ -83,28 +83,27 @@ def resolve(self, import_url):
         failed_urls = {}
         # trying to find a matching rule that can resolve this url
         for rule in self.rules:
-            key = rule.keys()[0]
-            value = rule.values()[0]
-            prefix = key
-            prefix_len = len(key)
+            # the validate method checks that the dict has exactly 1 element
+            prefix, value = list(rule.items())[0]
+            prefix_len = len(prefix)
             if prefix == import_url[:prefix_len]:
                 # found a matching rule
                 url_to_resolve = value + import_url[prefix_len:]
                 # trying to resolve the resolved_url
-                if url_to_resolve not in failed_urls.keys():
+                if url_to_resolve not in failed_urls:
                     # there is no point to try to resolve the same url twice
                     try:
                         return read_import(url_to_resolve)
-                    except DSLParsingLogicException, ex:
+                    except DSLParsingLogicException as ex:
                         # failed to resolve current rule,
                         # continue to the next one
-                        failed_urls[url_to_resolve] = str(ex)
+                        failed_urls[url_to_resolve] = ex
```

Comment:
this is only used for display, and will be repr'd by python anyway in
https://github.com/cloudify-cosmo/cloudify-dsl-parser/blob/f21d84140c2c581693961861a4618c0338e273e4/dsl_parser/import_resolver/default_import_resolver.py#L116

we should never use straight up `str()` like this because that's a good way to get those lovely UnicodeEncodeErrors